### PR TITLE
fix error with rdp scanníng

### PR DESCRIPTION
**Pull request by firefart**
There is no `match?` for Regexp

### CODE REVIEW

File: modules/auxiliary/scanner/rdp/rdp_scanner.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def rdp?
     sock.put(@probe)
     response = sock.get_once(-1)
     if response
-      if RDP_RE.match?(response)
+      if RDP_RE.match(response)
```
Comment (jhart-r7) [inline modules/auxiliary/scanner/rdp/rdp_scanner.rb]: This works fine in 2.4.1.  It won't work in 2.3 and perhaps earlier versions.  I thought msf was 2.4.x?

**Comment by jhart-r7** [comment]
@wvu-r7 sorry for stealing this from you.

File: modules/auxiliary/scanner/rdp/rdp_scanner.rb (hunk 1 above)
Comment (FireFart) [inline modules/auxiliary/scanner/rdp/rdp_scanner.rb]: Ruby 2.2 is still supported. Stumbled on this error on Kali. We recently had a problem with `Numberic positve?` dealing with the same problem: https://github.com/rapid7/metasploit-framework/pull/8792

**Comment by wvu-r7** [comment]
I forget if you can land things, @jhart-r7. If you can, please unassign me and take this. :)

**Comment by jhart-r7** [comment]
Got it, @wvu-r7 and @FireFart.  Thanks for the clarification, catch and fix.  Will land after testing.

**Comment by wvu-r7** [comment]
```
msf auxiliary(rdp_scanner) > ruby --version
[*] exec: ruby --version

ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
msf auxiliary(rdp_scanner) > run

From: /Users/wvu/metasploit-framework/modules/auxiliary/scanner/rdp/rdp_scanner.rb @ line 45 Msf::Modules::Mod617578696c696172792f7363616e6e65722f7264702f7264705f7363616e6e6572::MetasploitModule#rdp?:

    41: def rdp?
    42:   sock.put(@probe)
    43:   response = sock.get_once(-1)
    44:   if response
 => 45:     require 'pry'; binding.pry
    46:     if RDP_RE.match?(response)
    47:       # XXX: it might be helpful to decode the response and show what was selected.
    48:       print_good("Identified RDP")
    49:       return true
    50:     else
    51:       vprint_status("No match for '#{Rex::Text.to_hex_ascii(response)}'")
    52:     end
    53:   else
    54:     vprint_status("No response")
    55:   end
    56: end

[1] pry(#<Msf::Modules::Mod617578696c696172792f7363616e6e65722f7264702f7264705f7363616e6e6572::MetasploitModule>)> response
=> "\x03\x00\x00\x13\x0E\xD0\x00\x00\x124\x00\x02\x01\b\x00\x02\x00\x00\x00"
[2] pry(#<Msf::Modules::Mod617578696c696172792f7363616e6e65722f7264702f7264705f7363616e6e6572::MetasploitModule>)> RDP_RE
=> /^\x03\x00.{3}\xd0.{5}.*$/
[3] pry(#<Msf::Modules::Mod617578696c696172792f7363616e6e65722f7264702f7264705f7363616e6e6572::MetasploitModule>)> RDP_RE.match?(response)
=> true
[4] pry(#<Msf::Modules::Mod617578696c696172792f7363616e6e65722f7264702f7264705f7363616e6e6572::MetasploitModule>)> RDP_RE.match(response)
=> #<MatchData "\x03\x00\x00\x13\x0E\xD0\x00\x00\x124\x00\x02\x01\b\x00\x02\x00\x00\x00">
[5] pry(#<Msf::Modules::Mod617578696c696172792f7363616e6e65722f7264702f7264705f7363616e6e6572::MetasploitModule>)> 

[+] [redacted]:3389     - Identified RDP
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf auxiliary(rdp_scanner) > 
```

**Comment by jhart-r7** [comment]
Confirmed that this was a bug with 2.2.x and that with this fix it works in 2.2 - 2.4

**Comment by busterb** [comment]
this is silly, we added a gem that was supposed to make this something we didn't have to deal with, and recently I've hit it a half-dozen times. did we inadvertently remove the compat gem?

**Comment by busterb** [comment]
Yeah - 'match?' should be handled by the backports gem https://github.com/marcandre/backports - what's going on?

**Comment by jhart-r7** [comment]
@bcook-r7 I didn't know about our usage of `backports` but from what I've read just now, https://github.com/marcandre/backports/blob/7249b3afa7b6814b0026f4235ec6d3afeda4ad5c/lib/backports/2.4.0/regexp/match.rb, it seems like this should have never been an issue.

**Comment by busterb** [comment]
currently we `require 'backports`' - which I thought included everything. But it appears to only include features up to 2.1 (not very useful :)

https://github.com/marcandre/backports/blob/master/lib/backports.rb

to add the other stuff, one needs to specifically require it. I might submit a PR to upstream to make `require 'backports'` great again. Anyway, something to keep in mind next time, since match? actually is useful.

**Comment by busterb** [comment]
```
2.2.7 :002 > require 'backports/2.4.0/string/match'
 => true
2.2.7 :003 > "browncow".match?(/cow/)
 => true
2.2.7 :004 > "browncow".match(/cow/)
 => #<MatchData "cow">
```

**Comment by alrosenthal-r7** [comment]
# Release Notes

This fix resolves a backwards compatibility issue in the auxiliary/scanner/rdp/rdp_scanner module. The module is now compatible with Ruby 2.2 - 2.4.

**Pull request merged**